### PR TITLE
[batch] aggressively reduce cpus per PR

### PR DESCRIPTION
**Pull request by danking**
Screen Shot 2022-06-09 at 10 32 21 AM

### CODE REVIEW

**Comment by danking** [comment]
ugh that tail is all service backend stuff; it has terrible utilization.

**Comment by jigold** [comment]
Is part of the issue that the jobs are slow and there's a lot of them or is it the scheduler not keeping up? One hack we could consider is partial core JVM jobs. This would trivially get us 4x utilization.

**Comment by danking** [comment]
I'm not sure

**Comment by danking** [comment]
Screen Shot 2022-06-10 at 10 08 59 AM

Average utilization is marginally improved (maybe 2.5% -> 5%), but total number of wasted cores goes way down because the total number of cores is 16 rather than 16 * 8 = 128.

This also suggests the autoscaler/scheduler could be made substantially smarter.

**Comment by danking** [comment]
While fixing a bug in globbing I introduced a bug in non-globbed code. I'll fix that and then get this PR in.

**Comment by danking** [comment]
OK, I moved the file format test changes to https://github.com/hail-is/hail/pull/11906. This change can go in independently, but #11906 will even out the test job times and make developer experience better.

Service backend tests on #11904 which should be representative of a normal PR:

id | name | state | exit_code | duration
-- | -- | -- | -- | --
118 | test_hail_python_service_backend_0 | Success | Success 🎉 | 24 minutes
119 | test_hail_python_service_backend_1 | Success | Success 🎉 | 27 minutes
120 | test_hail_python_service_backend_2 | Success | Success 🎉 | 24 minutes
121 | test_hail_python_service_backend_3 | Success | Success 🎉 | 41 minutes
122 | test_hail_python_service_backend_4 | Success | Success 🎉 | 21 minutes

Service backend tests on this PR (albeit with #11906 which evens out test times):


id | name | state | exit_code | duration
-- | -- | -- | -- | --
118 | test_hail_python_service_backend_0 | Failed | Failure 🤷‍♀️ (1) | 31 minutes
119 | test_hail_python_service_backend_1 | Success | Success 🎉 | 31 minutes
120 | test_hail_python_service_backend_2 | Success | Success 🎉 | 28 minutes
121 | test_hail_python_service_backend_3 | Success | Success 🎉 | 33 minutes
122 | test_hail_python_service_backend_4 | Success | Success 🎉 | 26 minutes


I think there is almost no effect on service backend test times! We should really see if there's a way to improve the autoscaler & schedule to achieve this on its own.

File: batch/sql/set-test-and-dev-pools-to-8-core-max-2.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import os
+import asyncio
+from gear import Database
+
+
+async def main():
+    if os.environ['HAIL_SCOPE'] == 'deploy':
```
Comment (jigold) [inline batch/sql/set-test-and-dev-pools-to-8-core-max-2.py:7]: Can we add "dev" here?

**Review by jigold — DISMISSED** [review]

**Review by jigold — APPROVED** [review]

**Pull request merged**